posthog/mixins.py

```diff
@@ -23,4 +33,14 @@ def destroy(self, request, *args, **kwgars):
         self.perform_destroy(instance)
 
         report_user_action(request.user, f"{instance._meta.verbose_name} deleted", metadata)
+
+        if _should_save_historical_version(instance) and metadata:
```

Comment:
Can we specifically add to `FeatureFlagViewSet` rather than using a Mixin?

I don't see much benefit doing this, but makes the code non-obvious. Ideally I'd love to be able to look at FeatureFlagViewSet and see action create, update, destroy. This appears to be premature abstraction imo

Reply:
Have rejigged things a bit to keep the existing behaviour and move the destroy method onto the `FeatureFlagViewSet`